src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -2,18 +2,39 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.Runtime;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
-
+#if NATIVEAOT
 using Internal.Runtime;
+#else
+using System.Runtime.ExceptionServices;
+#endif
 
 // Disable: Filter expression is a constant. We know. We just can't do an unfiltered catch.
 #pragma warning disable 7095
 
+#if !NATIVEAOT
+namespace System
+{
+    public static partial class AppContext
+    {
+        internal static void OnFirstChanceException(object e)
+        {
+            FirstChanceException?.Invoke(AppDomain.CurrentDomain, new FirstChanceExceptionEventArgs((Exception)e));
+        }
+        internal static void OnUnhandledException(object e)
+        {
+            UnhandledException?.Invoke(AppDomain.CurrentDomain, new UnhandledExceptionEventArgs(e, true));
```

Comment:
There were tests failing without this being implemented here because they were not getting the notification, but it might be that it was just a point in time during the feature development and that now all unhandled exception end up going through the unmanaged code. I'll test it to see.